src/main/java/com/kodilla/ecommercee/service/OrderService.java

```diff
@@ -0,0 +1,67 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.Order;
+import com.kodilla.ecommercee.dto.OrderDto;
+import com.kodilla.ecommercee.exception.NoSuchIdException;
+import com.kodilla.ecommercee.mapper.OrderMapper;
+import com.kodilla.ecommercee.repository.OrderRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@AllArgsConstructor
+@Service
+public class OrderService {
+
+    private final OrderMapper mapper;
+    private final OrderRepository repository;
+
+    public List<OrderDto> getAll(){
+        return mapper.mapToOrderDtoList(repository.findAll());
+    }
+
+    public OrderDto getById(long id) throws NoSuchIdException{
+        Optional<Order> order = repository.findById(id);
+
+        if(order.isPresent()){
+            return mapper.mapToDto(order.get());
+        }else{
+            throw new NoSuchIdException();
+        }
+    }
+
+    @Transactional
+    public OrderDto create(final OrderDto orderDto){
```

Comment:
Ja bym tu dodał `src/main/java/com/kodilla/ecommercee/exception/CreatingObjectWithIdException.java` tak jak w mojej implementacji. Zapobiegnie to nadpisaniu istniejącego elementu (id>0), gdy ktoś poda już istniejące id - metoda kontrolera repozytorium save() służy do zapisywania i aktualizowania.